### PR TITLE
Numerous packages for security updates

### PR DESCRIPTION
**Pull request by montgomp**
A number of packages updated. Moved to `@azure/arm-xxx` packages required reworking client initializations and moving to `@azure/ms-rest-nodeAuth.TokenCredentialsBase`.

### CODE REVIEW

**Comment by formulahendry** [comment]
Please fix the compile issue in CI.

File: package.json [hunk 1]
```diff
@@ -790,52 +790,52 @@
...
+    "@types/mocha": "^8.0.2",
+    "@types/node": "^14.6.0",
+    "@types/node-fetch": "^2.5.7",
+    "@types/terser-webpack-plugin": "^4.1.0",
+    "@types/webpack": "^4.41.21",
+    "copy-webpack-plugin": "^6.0.3",
     "fail-on-errors-webpack-plugin": "^3.0.0",
-    "mocha": "^5.1.1",
-    "rimraf": "^2.6.2",
+    "keytar": "^6.0.1",
```
Comment (formulahendry) [inline package.json:804]: I think we could remove `keytar` from devDependencies

**Review by formulahendry — CHANGES_REQUESTED** [review]
Just one small comment. Others look good to me.

File: package.json (hunk 1 above)
Comment (montgomp) [reply package.json:804]: This is the issue that was failing the LInux build.

**Pull request merged**